crates/sim2h/src/job/pool.rs

```diff
@@ -7,30 +7,50 @@ pub(crate) struct Pool {
     job_send: crossbeam_channel::Sender<Box<dyn Job>>,
 }
 
+lazy_static! {
+    static ref SET_THREAD_PANIC_FATAL: bool = {
+        let orig_handler = std::panic::take_hook();
+        std::panic::set_hook(Box::new(move |panic_info| {
+            // invoke the default handler and exit the process
+            orig_handler(panic_info);
+            std::process::exit(1);
+        }));
+        true
+    };
+}
+
 impl Pool {
     pub(crate) fn new() -> Self {
+        // make sure if a thread panics, the whole process exits
+        assert!(*SET_THREAD_PANIC_FATAL);
```

Comment:
Would this get optimized away with a compiler flag? If so, intentional?

Reply:
I hope not! - the dereference to trigger the hook update is the whole point : )